### ISSUE TITLE
ASP.NET Web Forms Integration Guide Questions 

### ISSUE THREAD

**Issue by smarteez** [simpleinjector/SimpleInjector#491]
Hi 
I have been following the ASP.net Webforms Examples using the following link (https://simpleinjector.readthedocs.io/en/latest/webformsintegration.html ) from your documents and have some questions for stuff I am unsure about 

I split the code into class as follows 
1. PageInitializerModule 
2. Global.asmx
3. IocConfig ( this class I created the Bootstrap method ) 

**Question 1 : PageInitializerModule**
1. Where do I need to call this class? 
2. I saw the following code below where should I put it in my application?

[assembly: PreApplicationStartMethod(
    typeof(MyWebApplication.PageInitializerModule),
    "Initialize")]

****Question 2: IocConfig ( this class I created the Bootstrap method )**** 
I have an issue with the following line of code what do you advise I change I cant access the variable from the Gloal.asmx ( my class is the same as the example ) 
            Global.container = container;
**Whats I did is change the private to public is this correct** 

``` c#
public class Global : HttpApplication {
        private static Container container;

        public static void InitializeHandler(IHttpHandler handler) {
            container.GetRegistration(handler.GetType(), true).Registration
                .InitializeInstance(handler);
        }

        protected void Application_Start(object sender, EventArgs e) {
            Bootstrap();
        }
```

**Question 3:  Page** 
All the Objects are coming back null. can you advise why?

``` c#
[Import] public IUserRepository UserRepository { get; set; }
[Import] public IUserContext UserContext { get; set; }
```

**Comment by dotnetjunkie** [comment]
> PageInitializerModule: Where do I need to call this class?

That's what the `PreApplicationStartMethod` attribute is for. By placing it on the assembly level, ASP.NET will automatically call the configured "Initialize" method.

> where should I put it in my application?

Exactly as shown in the documentation. It is an assembly level attribute, which means it should be placed _outside_ a namespace, but you can place it in every C# file of the web project. Most convenient of course is to keep it in the same file as `PageInitializerModule`.

 > Question 2: IocConfig ( this class I created the Bootstrap method )

I'm not following that question.

>Page: All the Objects are coming back null. can you advise why?

That's because the `PageInitializerModule.Initialize` method wasn't called. Make sure you place the `PreApplicationStartMethod` correctly.

The funny thing though about this code sample is that it is just a complete copy-pastable example. My advice is to copy-paste the code as-is (with perhaps a rename of the `MyWebApplication` namespace) and run it to see if it works. After that, try to make changes step by step while ensuring that it keeps working.

**Comment by dotnetjunkie** [comment]
Do note that I made a little tweek to the [documentation](https://simpleinjector.readthedocs.io/en/latest/webformsintegration.html) making it a bit clear what external packages to use (both Simple Injector NuGet and the System.ComponentModel.Composition assembly), added some missing namespaces and updated the code examples to match the new Simple Injector v4 API.

**Comment by smarteez** [comment]
Hi 

I see you added this line 

container.Options.DefaultScopedLifestyle = new WebRequestLifestyle();

whats using must i used for WebRequestLifestyle?

**Comment by dotnetjunkie** [comment]
> whats using must i used for WebRequestLifestyle?

Could you rephrase this? I'm not sure what it is you want to know.

**Comment by smarteez** [comment]
ok sure 

I added the following line of code 

container.Options.DefaultScopedLifestyle = new WebRequestLifestyle();

WebRequestLifestyle is giving me an error is asking me to create a class is there a  reference I missing

I currently got the following should i chnage 

container.Options.DefaultScopedLifestyle = new AsyncScopedLifestyle();

container.Options.DefaultScopedLifestyle = new WebRequestLifestyle();

public static void Configure()
        {
            var container = new Container();

            container.Options.PropertySelectionBehavior = new ImportAttributePropertySelectionBehavior();

            container.Options.DefaultScopedLifestyle = new AsyncScopedLifestyle();

            RegisterRepositories(container);

            RegisterWebPages(container);
           

            Global.container = container;

            container.Verify();

           
        }

**Comment by smarteez** [comment]
thanks 

container.Options.DefaultScopedLifestyle = Lifestyle.Hybrid(
    defaultLifestyle: new WebRequestLifestyle(),
    fallbackLifestyle: new AsyncScopedLifestyle());

Hybrid section is giving me an error is asking me to change it to .... is this correct ?

container.Options.DefaultScopedLifestyle = Lifestyle.CreateHybrid(
    defaultLifestyle: new WebRequestLifestyle(),
    fallbackLifestyle: new AsyncScopedLifestyle());

**Comment by dotnetjunkie** [comment]
Yes, I'm sorry. That was a typo. `CreateHybrid` is correct.

**Comment by smarteez** [comment]
ok thanks Its working now 

Thank you so much for all your information
